### PR TITLE
file: Support macos for non-overwrite renaming

### PR DESCRIPTION
**Pull request by stanaka**
Add macos support for renaming without "overwrite" option.

### CODE REVIEW

File: src/file.rs [hunk 1]
```diff
@@ -38,7 +38,35 @@ fn rename(from: &Path, to: &Path, overwrite: bool) -> io::Result<()> {
...
+    } else {
+        Err(io::Error::last_os_error())
+    }
+}
+
+
+#[cfg(not(any(target_os = "linux", target_os = "macos")))]
 fn rename(from: &Path, to: &Path, overwrite: bool) -> io::Result<()> {
     use crate::util::die;
     if !overwrite {
```
Comment (cdown) [inline src/file.rs:72]: For the die!() below, change it to:

```rust
die!("Overwrite-free rename not implemented on this operating system. Use --overwrite.");
```

**Review by cdown — CHANGES_REQUESTED** [review]
This looks great, thanks. Maybe there's some opportunity for refactoring here, but it needs some thinking about anyway, so let's leave that for now.

Only one minor comment to address, and you need to run `cargo fmt`, then I'll merge :-)

**Comment by stanaka** [comment]
@cdown Thank you for the comment!
I added commits to address them. Please take a look on it.

**Comment by cdown** [comment]
Thank you! I added integration tests for overwrite/non-overwrite in 05ff1b046d04922c2292981484309b30754d8e0f, so there were conflicts. I've resolved the conflicts and tested that Mac passes the integration tests (see https://github.com/cdown/exifrename/actions/runs/11658829460/job/32458444391), so everything is great. :-)

Merged as d5dca9242d7df401db8eb13fa09cf8606e12bb2e. Thank you again!

**Comment by cdown** [comment]
1.1.0 released to crates.io with this feature. Thanks!

**Pull request closed without merging**